WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -208,49 +264,39 @@ private extension PointOfSaleDashboardViewModel {
             }
         }.assign(to: &$showsCardReaderSheet)
     }
-}
 
-import enum Yosemite.OrderAction
-import struct Yosemite.Order
-@MainActor
-private extension PointOfSaleDashboardViewModel {
-    private func createOrUpdateTestOrder() async throws -> Order {
-        if let order {
-            let updatedOrder = try await update(order: order, amount: formattedOrderTotalPrice ?? "15.00")
-            self.order = updatedOrder
-            return updatedOrder
-        } else {
-            let order = try await createTestOrder(amount: formattedOrderTotalPrice ?? "15.00")
+    @MainActor
+    private func syncOrder(for cartProducts: [CartItem]) async {
+        guard isSyncingOrder == false else {
+            return
+        }
+        isSyncingOrder = true
+        let cart = cartProducts
+            .map {
+                POSCartItem(itemID: nil,
+                            product: $0.item,
+                            quantity: Decimal($0.quantity))
+            }
+        defer {
+            isSyncingOrder = false
+        }
+        do {
+            isSyncingOrder = true
+            let order = try await orderService.syncOrder(cart: cart,
+                                                              order: order,
+                                                              allProducts: items)
             self.order = order
```

Comment:
```suggestion
            isSyncingOrder = true
            let order = try await orderService.syncOrder(cart: cart,
                                                              order: order,
                                                              allProducts: items)
            self.order = order
            isSyncingOrder = false
```

If we keep the defer in place, it means that we don't end up forgetting when errors are thrown... but we could repeat the call in the catch block if you prefer.